Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3472,7 +3473,11 @@ else if (thisReq_ instanceof ServerRequestLogout) {
                 } catch (JSONException ex) {
                     ex.printStackTrace();
                 }
+                if (thisReq_ instanceof ServerRequestInitSession) {
+                    branchUniversalReferralInitWrapper_ = null;
```

Comment:
I got it. But what if Auto Session Management kicks in, triggering a new session next time the app is opened. The callback won't be called because you've nullified it? I didn't check this path but if you did, I'm 👍